### ISSUE TITLE
cmd/cover: panic: overlapping edits

### ISSUE THREAD

**Issue by Cosby86** [golang/go#23927]
Hello all with the latest RC if i run tests all is ok, if I run them with coverage I run into this error:
```
cover poc/all_test
panic: overlapping edits: [4371,4375)->"", [4372,4372)->"_cover_atomic_.AddUint32(&GoCover_2_646536633930623366336461.Count[26], 1);"

goroutine 1 [running]:
cmd/internal/edit.(*Buffer).Bytes(0xc0421223c0, 0xc0420421e0, 0x63b4a0, 0xc0420ee580)
	C:/workdir/go/src/cmd/internal/edit/edit.go:79 +0x51c
main.annotate(0xc042044060, 0x53)
	C:/workdir/go/src/cmd/cover/cover.go:332 +0x46f
main.main()
	C:/workdir/go/src/cmd/cover/cover.go:88 +0x187 ```

**Comment by ianlancetaylor** [comment]
Can you tell us how to recreate the problem?

**Comment by Cosby86** [comment]
Actually i have 340 packages with tests and only one is having this issue. It seems the file is not different from the others. I still have to identify what's causing the issue or how to replicate it. I'll let you know very soon.

**Comment by Cosby86** [comment]
I worked and found what's causing the issue.  I have this function (it is only an example but it's recreating the issue). With the function below I don't have the issue:

```
func updateItemsChildrenKeys(itemsColl data.PhysicalItemsCollection, items []data.Item, request *command.InternalRequest, reply *command.Reply) errors.Err {
    up := itemsColl.StartUpdate()
    defer up.EndUpdate()

    for _, item := range items {
        _, _ = up.ByKeyForUpdate(item.Key())
        if item.ParentKey() != nil {

        }

    }

    return nil
}
```

but when I add the else the issue appears:
```
func updateItemsChildrenKeys(itemsColl data.PhysicalItemsCollection, items []data.Item, request *command.InternalRequest, reply *command.Reply) errors.Err {
    up := itemsColl.StartUpdate()
    defer up.EndUpdate()

    for _, item := range items {
        _, _ = up.ByKeyForUpdate(item.Key())
        if item.ParentKey() != nil {

        }**else{
            
        }**

    }

    return nil
}
```

**Comment by Cosby86** [comment]
I was able to replicate the issue with this code too:

```
if len(attributes) > 0{
			if item != nil{
                            //DO SOMETHING
			}
		}else{
                    //DO SOMETHING
		}
```

**Comment by Cosby86** [comment]
with this code works:

```

**Comment by ianlancetaylor** [comment]
Thanks.  For clarity, can you show us a complete, self-contained example, along with the commands you use to recreate the problem?  That would save time on our end.  Thanks.

**Comment by Cosby86** [comment]
Atttached .go file and its related tests. It is veri very very very simple but it reproduce the issue. Let me know

[test.zip](https://github.com/golang/go/files/1761647/test.zip)

**Comment by ianlancetaylor** [comment]
Thanks.  The problem only arises because your main.go file has not been through gofmt.  If I run gofmt on the file, the problem disappears.

The problem is specifically `}else{`.  If I change that to `} else{` (adding a space after the `}`, as gofmt does) then the problem does not occur.

main.go:

```Go
package main

func main(){
	var attributes []string
	var item interface{}

	if len(attributes) > 0{
		if item != nil{
			//DO SOMETHING
		}
	}else{
		//DO SOMETHING
	}
}
```

main_test.go:

```Go
package main

import (
	"testing"
)

func TestMain(t *testing.T) {
	main()
}
```

go test -cover

```
# cover _/tmp/x/src/a
panic: overlapping edits: [142,146)->"", [143,143)->"GoCover_0_366235663035346337313436.Count[3] = 1;"

goroutine 1 [running]:
cmd/internal/edit.(*Buffer).Bytes(0xc0000f2c00, 0xc0000f0120, 0x6216a0, 0xc000110300)
	/home/iant/go/src/cmd/internal/edit/edit.go:79 +0x515
main.annotate(0x7ffeec2bb74a, 0x14)
	/home/iant/go/src/cmd/cover/cover.go:332 +0x468
main.main()
	/home/iant/go/src/cmd/cover/cover.go:88 +0x180
FAIL	_/tmp/x/src/a [build failed]
```

**Comment by Cosby86** [comment]
😮 got it

**Comment by andybons** [comment]
CL 98935 OK for Go 1.10.1

**Comment by AgustinBanchio** [comment]
I had this problem with go 1.10.3
Adding the spaces in the else clauses fixed it.
I was testing the coverage with GoLand so it might be their tools aren't updated

**Comment by andybons** [comment]
@AgustinBanchio if you test without GoLand can you still reproduce the issue?

**Comment by ianlancetaylor** [comment]
@AgustinBanchio This problem is fixed in 1.10.1.  If you are seeing the exact same problem, then you are most likely running an older version of the toolchain.  If you are sure that you are not, then you have found a different problem.  Please open a new issue with reproduction instructions.  Thanks.
